### PR TITLE
update logtools to support verbosity Zero Check

### LINKED ISSUE #112206: update logtools to support verbosity Zero Check
### What would you like to be added?

as discussed in https://github.com/kubernetes-sigs/logtools/issues/2 and https://github.com/kubernetes/kubernetes/pull/111708, we should add verbosity Zero Check in logtools

### Why is this needed?

because with V(0) and without V(0) semantically equivalent and V(0) add some (small) overhead
E.g:
`klog.V(0).InfoS` and `klog.InfoS` are equivalent. We don't need to use klog.V(0).InfoS

### PR DESCRIPTION
**Pull request by yangjunmyfm192085**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
as discussed in https://github.com/kubernetes-sigs/logtools/issues/2 and https://github.com/kubernetes/kubernetes/pull/111708, we should add verbosity Zero Check in logtools
#### Which issue(s) this PR fixes:

Fixes #112206

#### Does this PR introduce a user-facing change?

```release-note
"NONE"
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
"NONE"
```

### CODE REVIEW

File: hack/tools/go.mod
Comment (pohly) [inline hack/tools/go.mod:14]: Let's do a formal release of logtools first. For that I am waiting for https://github.com/kubernetes-sigs/logtools/pull/1

**Comment by yangjunmyfm192085** [comment]
/hold
only for test, don't merge

**Review by yangjunmyfm192085 — COMMENTED** [review]
/hold
only for test, don't merge

**Review by pohly — CHANGES_REQUESTED** [review]
I released logtools v0.3.0. Can you update Kubernetes to that tagged release?

**Comment by yangjunmyfm192085** [comment]
> I released logtools v0.3.0. Can you update Kubernetes to that tagged release?

ok, Let me update to v0.3.0.
But before that, I think we need to merge https://github.com/kubernetes/kubernetes/pull/112671 and https://github.com/kubernetes/kubernetes/pull/112672

**Review by yangjunmyfm192085 — COMMENTED** [review]
/sig instrumentation

**Comment by pohly** [comment]
We probably need a list of all pending PRs that address the issues found by this tool update. Here are some that I am aware of:
- https://github.com/kubernetes/kubernetes/pull/112670
- https://github.com/kubernetes/kubernetes/pull/110747

**Review by yangjunmyfm192085 — COMMENTED** [review]
/unhold

**Comment by yangjunmyfm192085** [comment]
/cc @pohly logcheck will still check for errors.
I'll analyze it later

**Comment by pohly** [comment]
Oh, I had forgotten about this PR and did my own with the same logtools update.

The errors are keys in existing code that violate our guidelines and now trigger warnings. I fixed them in https://github.com/kubernetes/kubernetes/pull/115060.

**Review by yangjunmyfm192085 — COMMENTED** [review]
.Closed due to PR expiration

**Pull request closed without merging**
